src/Subscription.spec.ts

```diff
@@ -0,0 +1,230 @@
+      } as unknown as SubscriptionOptions;
+      const tryValidate = () => validate('id', badConfig);
+      assert.throws(tryValidate, Error, 'Subscription handling option onCreate must be a function or an array of functions');
+
+      const badConfig2 = {
+        onCreate: false,
+        onConfigure: NOOP_FXN,
+        onDeleted: NOOP_FXN,
+      } as unknown as SubscriptionOptions;
+      const tryValidate2 = () => validate('id', badConfig2);
+      assert.throws(tryValidate2, Error, 'You must supply a middleware for onCreate');
+
+      const badConfig3 = {
+        onCreate: NOOP_FXN,
+        onConfigure: [NOOP_FXN, 'not a middleware fxn'],
+      } as unknown as SubscriptionOptions;
+      const tryValidate3 = () => validate('id', badConfig3);
+      assert.throws(tryValidate3, Error, 'Subscription handling option(s) supplied as arrays must contain only functions');
+    });
+  });
+
+  describe('getMiddleware()', () => {
+    it('should NOT call next() for notification_subscription_* event', async () => {
+      const ws = new Subscription('id', MOCK_SUBSCRIPTION_CONFIG_SINGLE as unknown as SubscriptionOptions);
+      const middleware = ws.getMiddleware();
+      const mockValidCreateArgs = mockNotificationSubscriptionCreateRequested(true) as unknown as
+      SlackSubscriptionMiddlewareArgs & AllMiddlewareArgs;
+      const mockNext = sinon.spy();
+      mockValidCreateArgs.next = mockNext;
+
+      await middleware(mockValidCreateArgs);
+      assert(mockNext.notCalled);
+    });
+    it('should call next() if NOT a notification_subscription_* event', async () => {
+      const ws = new Subscription('id', MOCK_SUBSCRIPTION_CONFIG_SINGLE as unknown as SubscriptionOptions);
+      const middleware = ws.getMiddleware();
+      const mockInvalidCreateArgs = mockNotificationSubscriptionCreateRequested(false) as unknown as
+      SlackSubscriptionMiddlewareArgs & AllMiddlewareArgs;
+      const mockNext = sinon.spy();
+      mockInvalidCreateArgs.next = mockNext;
+
+      await middleware(mockInvalidCreateArgs);
+      assert(mockNext.called);
+    });
+  });
+
+  describe('isSubscriptionInteraction', () => {
+    it('returns true for valid types', async () => {
+      const { isSubscriptionInteraction } = await importSubscription();
+      const mockValidCreateArgs = mockNotificationSubscriptionCreateRequested(true) as unknown as
+      SlackSubscriptionMiddlewareArgs & AllMiddlewareArgs;
+      assert.isTrue(isSubscriptionInteraction(mockValidCreateArgs));
+    });
+    it('returns false for invalid types', async () => {
+      const { isSubscriptionInteraction } = await importSubscription();
+      const mockInvalidCreateArgs = mockNotificationSubscriptionCreateRequested(false) as unknown as
+      SlackSubscriptionMiddlewareArgs & AllMiddlewareArgs;
+      assert.isFalse(isSubscriptionInteraction(mockInvalidCreateArgs));
+    });
+  });
+
+  describe('matchesConstraints()', () => {
+    // TODO: Implement
```

Comment:
Added a test stub here when matchesConstraints is able filter events that match custom_id constraint